source/api_cc/src/DeepPotPT.cc

```diff
@@ -47,6 +47,12 @@ void DeepPotPT::init(const std::string& model,
     std::cout << "load model from: " << model << " to gpu " << gpu_rank
               << std::endl;
   }
+  int gpu_num = -1;
+  DPGetDeviceCount(gpu_num);
+  if (gpu_id > gpu_num) {
+    throw deepmd::deepmd_exception(
+        "current rank" + gpu_id + "is larger than the number of gpu" + gpu_num);
+  }
```

Comment:
We should use `gpu_id %= gpu_num` like TF

https://github.com/deepmodeling/deepmd-kit/blob/b875ea8f6661b6e1567537ead7e2b4a8b14ea113/source/api_cc/src/DeepPotTF.cc#L447-L449